### ISSUE TITLE
Rename members from referenced assemblies

### ISSUE THREAD

**Issue by AkyrosXD** [0xd4d/dnlib#302]
This is not actually an issue with the library, but most likely a question.
I am not sure if this has been mentioned before, but I am having a small problem.

I want to rename a member from a referenced assembly but when I try to do that, the memeber which is used in the main assembly stops working. I've been trying to fix this for so long and looking around on Google for solutions but I always find a dead end.

Code in main assembly:
```CSharp
public class SomeClass : MonoBehaviour
{
    public void Start()
    {
        int sum = ExamplePluginClass.Sum(5, 10);
        Debug.Log(sum);
    }
}
```

Code in referenced assembly:
```CSharp
public class ExamplePluginClass
{
    public static int Sum(int a, int b)
    {
        return a + b;
    }
}
```
I load the example plugin and rename all the types and then write it. When I open the plugin in dnspy, everything looks just like they should be. But, here is when the problem comes.

After renaming `ExamplePluginClass.Sum`, it doesn't work anymore. If I open the main assembly in dnspy, go to `SomeClass.Start` it will show the original names but not the changed one and when you click on the members, it won't redirect you to them as it should. I hope my problem is clear to understand and if there is a solution to this problem, please let me know.

**Comment by 0xd4d** [comment]
Sounds like you only rename the types in the original assembly but never rename any references to that type in other assemblies. You should open all assemblies that reference the target assembly and rename those references then save everything. You can have a look at de4dot's renamer, it uses dnlib's `MemberFinder` to find all refs.

**Comment by AkyrosXD** [comment]
Ok, I took a look at it and I kinda did what I wanted to do. But I am now stuck at something.

I managed to rename referenced types without any problem and now I wanna do the same with methods, fields, properties and events. 

But when I wanna check if the method is a constructor by using `MethodDef.IsConstructor` to exclude it for renaming I can't really find a way to do that.

I tried casting but does not work so I tried`MemberRef.ResolveMethod` but it always returns `null`.
